x/zkauth/keeper/keeper.go

```diff
@@ -105,18 +90,61 @@ func (k Keeper) ParseJWKs(byteArray []byte) (jwks []types.JWK, err error) {
 	return jwks, nil
 }
 
-func (k Keeper) GetJWK(endpoint string) (*http.Response, error) {
+func (k Keeper) FetchJWK(endpoint, nodeHome string, name types.OidcProvider) error {
 	req, err := http.NewRequest("GET", endpoint, nil)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	client := new(http.Client)
 	resp, err := client.Do(req)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	return resp, nil
+	targetFile := filepath.Join(nodeHome, k.CreateJWKFileName(name))
+	file, err := os.Create(targetFile)
+	if err != nil {
+		return err
+	}
+
+	_, err = io.Copy(file, resp.Body)
+	if err != nil {
+		return err
+	}
+
+	dataBytes, err := os.ReadFile(targetFile)
```

Comment:
How about parsing the resp as is?

Reply:
I changed the order of assigning to dataBytes and use WriteFile.

Reply:
The key is held in both the jwks field and the file, so I don't think it is necessary to save it to the file. What do you think?
https://github.com/Finschia/finschia-sdk/pull/1249/files#diff-2a2241d04eae2d37982aafdf8f50d27ae141f623782c48ee72fbe125ece11fc7R23

Reply:
I thought about it again, and it doesn't seem necessary to be stored in both as you say. I stopped saving locally.